### DISCUSSION TITLE
How to set the non-penetrating wall condition?

### DISCUSSION THREAD

**Discussion by YuhangNIU** [idaholab/moose#18188] [Q&A Modules: Navier-Stokes] [answered]
Hi all,

Can anybody kindly tell me how to set **the non-penetrating wall condition** in **one-dimensional flow**?
Just let **v=0** at the boundary?

Thank you!
Yuhang

**Comment by GiudGiud** [comment]
Hi Yuhang

Where is the wall in your 1D system? One of the edges?
Which formulation are you using? Finite element or finite volume? Incompressible or compressible?

Guillaume

**Reply by YuhangNIU** [reply]
Hi Guillaume,

Thank you for your reply. 
The walls are on both edges. The finite element formulation is used in the compressible problem (Sedimentation problem of two-phase fluid).

**Reply by GiudGiud** [reply]
I think this one
https://mooseframework.inl.gov/source/bcs/NSImposedVelocityBC.html
is the right one for FE and compressible. 
It might depend on your formulation as well. Which variables are you using, conservative or primitive ? (rho, rho u, rho E) or (P, u, T) ?
EDIT: this is for single phase problems btw. 

@lindsayad will know better either way

**Reply by WilkAndy** [reply]
You mentioned sedimentation: does this mean you're using PorousFlow?   Please ignore if this just causes confusion.

**Reply by YuhangNIU** [reply]
The two-fluid two-phase model (six-equation) was built in MOOSE, in which the variables I used are (p, ul, ug,α, Tl, Tg). And I did not use the NS  or PorousFlow module in MOOSE, all the system was developed in my own APP based on MOOSE.

**Comment by lindsayad** [comment, marked as answer]
The answer to this depends on a couple things. Are you using finite elements or finite volumes? Are you integrating the continuity equation by parts/are you applying the divergence theorem? If yes, then a natural boundary condition on the mass equation will already have you most of the way there for a *non-penetrating* condition. If you also want a zero tangential velocity (which it seems like you do) then we'll have to do more than for just a non-penetrating condition.

**Reply by GiudGiud** [reply]
It's finite element, and it's a 1D system so they're only trying to get 0 normal velocity.
By natural BC do you mean a BC that mimics the action of the kernel? Or a 0-residual BC? I think the latter, just making sure

**Reply by lindsayad** [reply]
I should read better

**Reply by lindsayad** [reply]
DirichletBCs with `value = 0` for the velocities would definitely do the trick.

**Reply by lindsayad** [reply]
> Or a 0-residual BC?

Yes I was referring to this for natural BC

**Reply by YuhangNIU** [reply]
Thanks for the help. Much appreciated!
